custom/icds_reports/utils/usage_data_pull.py

```diff
@@ -0,0 +1,88 @@
+from custom.icds_reports.models.util import ICDSAuditEntryRecord
+from custom.icds_reports.models.aggregate import  AggAwc
+import re
+from datetime import date
+from corehq.apps.users.models import CommCareUser
+import csv
+
+
+
+
+user_objects = {}
+dashboard_uname_rx = re.compile(r'^\d*\.[a-zA-Z]*@.*')
+
+
+date_setp = [(date(2019,9,2),date(2019,9,8)),(date(2019,9,9),date(2019,9,15)),(date(2019,9,16),date(2019,9,22)),(date(2019,9,23),date(2019,9,29)),(date(2019,9,30),date(2019,9,30))]
+dates_oct = [(date(2019,10,1), date(2019,10,6)),(date(2019,10,7),date(2019,10,13)),(date(2019,10,14),date(2019,10,20)),(date(2019,10,21),date(2019,10,27)),(date(2019,10,28),date(2019,11,3))]
+
+
+def get_user_obj(username):
+    if username in user_objects:
+        return user_objects[username]
+    else:
+        return CommCareUser.get_by_username(username)
+
+
+def get_launched_locations(month):
+
+    x = AggAwc.objects.filter(month=month,aggregation_level=3,num_launched_awcs__gt=0).values('state_id','district_id','block_id')
+
+    launched_states = []
```

Comment:
This is only used as the argument to an `in` query. They would be way more efficient as `set`s